plugins/experimental/certifier/certifier.cc

```diff
@@ -0,0 +1,739 @@
+  {
+    TSMutexLock(list_mutex);
+    auto iter = cnDataMap.find(commonName);
+    if (iter != cnDataMap.end()) {
+      std::swap(localQ, iter->second->vconnQ);
+      iter->second->ctx    = std::move(ctx);
+      iter->second->cert   = std::move(cert); ///< We might not need cert, can be easily removed
+      iter->second->wontdo = wontdo;
+    }
+    TSMutexUnlock(list_mutex);
+  }
+
+  // Prepend to the LRU list
+  void
+  prepend(SslData *data)
+  {
+    TSMutexLock(list_mutex);
+    std::unique_ptr<SslData> local = nullptr;
+    if (data != nullptr) {
+      // If data is the most recent node in the list,
+      // we leave it unchanged.
+      if (head != data) {
+        // Remove data from the list if it is in
+        bool present = false;
+        if (data->prev != nullptr) {
+          data->prev->next = data->next;
+          if (tail == data) {
+            tail = data->prev;
+          }
+          present = true;
+        }
+        if (data->next != nullptr) {
+          data->next->prev = data->prev;
+          present          = true;
+        }
+        data->prev = nullptr;
+        data->next = head;
+        if (data->next != nullptr) {
+          data->next->prev = data;
+        }
+        head = data;
+        if (tail == nullptr) {
+          tail = data;
+        }
+
+        // If node is already in the list, prepend simply changes order without resizing
+        if (!present) {
+          // Remove oldest node if size exceeds limit
+          if (++size > limit) {
+            TSDebug(PLUGIN_NAME, "Removing %s", tail->commonName.c_str());
+            auto iter = cnDataMap.find(tail->commonName);
+            if (iter != cnDataMap.end()) {
+              local = std::move(iter->second); // copy ownership
+              cnDataMap.erase(iter);
+            }
+            if ((tail = tail->prev) != nullptr) {
+              tail->next = nullptr;
+            }
+            size -= 1;
+          }
+        }
+      }
+    }
+    TSDebug(PLUGIN_NAME, "%s Prepend to LRU list...List Size:%d Map Size: %d", data->commonName.c_str(), size,
+            static_cast<int>(cnDataMap.size()));
+
+    TSMutexUnlock(list_mutex);
+  }
+
+  // Remove list node
+  void
+  remove(SslData *data)
+  {
+    std::unique_ptr<SslData> local;
+    TSMutexLock(list_mutex);
+    if (data != nullptr) {
+      if (data->prev != nullptr || data->next != nullptr || head == data) {
+        if (data->prev != nullptr) {
+          data->prev->next = data->next;
+        }
+        if (data->next != nullptr) {
+          data->next->prev = data->prev;
+        }
+        if (head == data) {
+          head = data->next;
+        }
+        if (tail == data) {
+          tail = data->prev;
+        }
+        data->prev = nullptr;
+        data->next = nullptr;
+        size -= 1;
+        auto iter = cnDataMap.find(tail->commonName);
+        if (iter != cnDataMap.end()) {
+          local = std::move(iter->second); // copy ownership
```

Comment:
What is the point of this? Isn't the `SslData` in `local` lost anyway when this goes out of scope?